File: desafio0.js
// Codificador Cesar

let alphabet = "ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
let message = "YO HE LOGRADO ENCENDER UNA CERILLA";
let N = 2;
let key1 = 2;
let key2 = 3;
let response = "";

message = message.split(/\s+/).join('');


for(let i=0; i<message.length; i+=N){
  response += alphabet[(alphabet.indexOf([message[i]])+key1)%alphabet.length];
  if(message.length-1 >= i+1){
    response += alphabet[(alphabet.indexOf([message[i+1]])+key2)%alphabet.length];
  }
}
    
  
console.log(response);

File: desafio0.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

string alphabet = "ABCDEFGHIJKLMNÑOPQRSTUVWXYZ";
string message = "YO HE LOGRADO ENCENDER UNA CERILLA";
#define N 2
int key1 = 2;
int key2 = 3;
string response = "";

int main() {
    message.erase(remove_if(message.begin(), message.end(), isspace), message.end());
    
    for(int i=0; i<message.size(); i+=N){
        int index = alphabet.find(i);
        response += alphabet[(index + key1) % alphabet.size()];

        if(message.size()-1 >= i+1){
            response += alphabet[(index + key2) % alphabet.size()];
        }
    }

    cout << response << '\n';
    return 0;
}